tests/SpecTests/TransactionsSpecTest.php

```diff
@@ -128,17 +132,17 @@ public static function assertCommandMatches(stdClass $expected, stdClass $actual
      */
     public function testTransactions(stdClass $test, array $runOn = null, array $data, $databaseName = null, $collectionName = null)
     {
+        $this->skipIfTransactionsAreNotSupported();
```

Comment:
Shouldn't we be able to rely on the `runOn` requirements to enforce this? I don't believe we used the common skip method in the change stream spec test runner, which is otherwise used by `WatchFunctionalTest`.

Reply:
The change stream spec tests only run on the `replicaset` topology, while `WatchFunctionalTest` skips tests if change streams are not supported (which checks whether a sharded cluster uses replica sets in the background).

The spec test can't specify to only run tests on a sharded cluster backed by replica set, as the topology will be `sharded` in both cases. The `topology` key of the `runOn` object is an array with the requirement being that the topology in use is listed in the topology array. IMO, this is a shortcoming of the current `runOn` syntax, but then from a topology perspective there is no difference between a sharded cluster using standalone servers and a cluster using replicasets.

FWIW, I added this skip after encountering test errors on a sharded cluster with standalone servers as that does not support transactions. This is due to the `sharded` topology requirement also matching for that cluster type. It could be that other drivers are not testing that particular use-case, which is why this went uncaught before.

Reply:
There is one change stream spec test that runs on a sharded cluster ([here](https://github.com/mongodb/specifications/blob/4d5c5c78f68eb1c3c1a4462e197fdc5fd2312bf0/source/change-streams/tests/change-streams-errors.yml#L54)), but I'm not sure if that error test even gets as far as caring that the sharded cluster may not be backed by a replica set.